src/utils/units.cpp

```diff
@@ -0,0 +1,100 @@
+//========================================================================================
+// (C) (or copyright) 2024. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+
+#include "units.hpp"
+
+namespace ArtemisUtils {
+
+Units::Units(ParameterInput *pin) {
+  unit_system_ = pin->GetOrAddString("artemis/units", "type", "code");
```

Comment:
I'd slightly prefer the input block be `<units>` and not `<artemis/units>` since we don't really have `artemis/*` blocks.

Reply:
My feeling was that `artemis/units` is the appropriate scope for these parameters, since 1) the units are used artemis-wide 2) other parthenon codes might have their own notion of `units` input parameters and this would avoid crosstalk between packages in different codes. 

We already have `artemis/` parameters so I don't see a style issue with having blocks as well there, but if you foresee issues with doing this lets keep discussing.